Source/WebCore/platform/graphics/ImageBuffer.h

```diff
@@ -161,6 +161,7 @@ class ImageBuffer : public ThreadSafeRefCounted<ImageBuffer, WTF::DestructionThr
     RenderingPurpose renderingPurpose() const { return m_parameters.purpose; }
     PixelFormat pixelFormat() const { return m_parameters.pixelFormat; }
     const ImageBufferBackend::Parameters& parameters() const { return m_parameters; }
+    virtual bool isRemote() const { return false; }
```

Comment:
I am struggling to balance between making this change small enough to be reviewable, but also making sure it works.

The existing state is that remote ImageBuffers only really work on the main-thread, but we let JS postMessage them into workers, and we just use `DestructionThread::Main` to make sure they get sent back if they get destroyed. Since OffscreenCanvas is disabled, there's no Web APIs that can use them on a worker, so they can exist on a worker without breaking anything.

This patch makes workers have their own remote rendering backend proxy, so that remote ImageBuffers created on a worker are now actually usable. Unfortunately, we now need to do something different when we copy an ImageBuffer between threads, so this patch forces read back into a local unaccelerated buffer.

I added `isRemote` as a temporary hack to know when to do this, to avoid regressing performance for Gtk (which does have offscreen canvas enabled).

Bug 244830 is going to add proper support for transferring ImageBuffers between threads (and changing them to use the appropriate remote rendering backend proxy), and I can remove `isRemote` again as part of that.

Reply:
The transfer changes are here: https://github.com/mattwoodrow/WebKit/commit/ee88514e6c393d65f8b47512aaf53b6582cd02e38c

Still needs some cleanup, but we won't need `isRemote` with those.